### DISCUSSION TITLE
Customise parameters used to generate sidebar toctree

### DISCUSSION THREAD

**Discussion by alexisthual** [pradyunsg/furo#354] [Potential Issue] [answered]
Currently, furo overwrites rst files' toctree parameters when it generates the sidebar toctree: https://github.com/pradyunsg/furo/blob/d6b217f8bb021f137928aa30f1b41e829e87d73f/src/furo/__init__.py#L97-L110

We at [nilearn](https://github.com/nilearn/nilearn) plan on using furo, but would like to keep the original parameters we set in the rst files.
What do you think is the best way for us to achieve this? Would it make sense that furo removes this parameters altogether by default and enable setting them in `conf.py`? This is actually what we would need as different sections of our toctree have different parameters.

Thank you for the very nice theme and maintenance anyways :smile:

**Comment by pradyunsg** [comment, marked as answer]
Hiya! Please see https://github.com/pradyunsg/furo/discussions/117#discussioncomment-555756. :)

Basically, it's an opinionated choice to ensure that the entire site structure is presented in the sidebar.

**Reply by alexisthual** [reply]
It's very nice that the theme comes with opinionated defaults, but I think there are some cases where users might need to overwrite these.
For instance, `titles_only=True` actually filters out part of our site structure, and so does `max_depth=2` (which, in our case, we need to increase).

**Reply by pradyunsg** [reply]
See https://github.com/pradyunsg/furo/discussions/146#discussioncomment-722198 for the rationale behind `titles_only=True`. There is no way to change that. There is also no `max_depth=2` in this codebase, as far as I know.

**Reply by pradyunsg** [reply]
See also https://github.com/pradyunsg/furo/discussions/155#discussioncomment-889767.

**Reply by pradyunsg** [reply]
And, finally, https://github.com/pradyunsg/furo/issues/79#issuecomment-774428681.

**Reply by alexisthual** [reply]
> See [#146 (comment)](https://github.com/pradyunsg/furo/discussions/146#discussioncomment-722198) for the rationale behind `titles_only=True`. There is no way to change that. There is also no `max_depth=2` in this codebase, as far as I know.

Oops, my bad about `max_depth`.

I understand your position, and think it makes sense to expose the website's true structure.
In my case, I changed the site's structure to obtain what I wanted.
However, I still think it would be nice that users can tweak this, and that the theme does not enforce it. As a user, what I want from furo is clean css (I really love the design :smile: ), but I don't want it to make other strong choices for me.
